### DISCUSSION TITLE
White padding while resizing

### DISCUSSION THREAD

**Discussion by jpomykala** [fabricjs/fabric.js#7791] [Q&A] [answered]
Hey! I'm struggling with this a few weeks now. I guess this might be connected with cache (copy from smaller canvas to the bigger one?) so far I tried to set `noScaleCache: true` and `objectCaching: false` but it doesn't help.

I would like to get rid of this white spaces while resizing.

https://user-images.githubusercontent.com/1548104/157454209-b9c68ca6-45d2-4f73-b2d6-dca6ea3efa50.mov

I'm using a 'object:scaling' hook in order to calculate rectangle width and height using `scaleX` and `scaleY`  

```
canvas.on('object:scaling', (options: IEvent) => {

      const target = options.target!;
      const selectedType = target.type || "";
      if (!options.target) {
        return;
      }

      if (selectedType === ComponentType.RECTANGLE) {
        options.target.width = Math.round(options.target.width! * options.target.scaleX!);
        options.target.height = Math.round(options.target.height! * options.target.scaleY!);
        options.target.scaleX = 1;
        options.target.scaleY = 1;
        options.target.dirty = true;
      }
    });
```

**Comment by ShaMan123** [comment, marked as answer]
Try using `object:modified` hook

**Reply by jpomykala** [reply]
Thank you, it worked!

**Reply by ShaMan123** [reply]
Great! 
If must resize on every change and not use scaling you can try
https://github.com/fabricjs/fabric.js/blob/2eabc92a3221dd628576b1bb029a5dc1156bdc06/src/mixins/default_controls.js#L101

**Reply by jpomykala** [reply]
Thank you for this! I have been using Fabric for ~4/5 months now, I've been looking into Fabric source code many times, but I never looked into this file. ☺️

**Reply by ShaMan123** [reply]
I know what you mean.
Hopefully the JS/TS rewrite will change this awful DX
